### ISSUE TITLE
Related objects not saving when saving the main object

### ISSUE THREAD

**Issue by jaysonng** [parse-community/Parse-SDK-iOS-OSX#1509]
I have a Tag Object and a Post object. Both of  which are a subclass of PFObject.

Given this code:
```
var post = Post()
let tagsRelation = post.relation(forKey: "tags")
let testTag = Tag()
testTag.tag = "testTag"
tagsRelation.add(Tag(withoutDataWithObjectId: "0ngpdctMer"))
tagsRelation.add(testTag)

```

then saving the post object

**Comment by noobs2ninjas** [comment]
Hey @jaysonng!

Correct me if I'm wrong, what you are doing is a PFRelation column or a one-to-many relationship where Post has a PFRelation called "tags" that points to many Tag objects. Right? 

Assuming this is correct, currently this SDK only automatically saves new one-to-one aka "Pointer" objects. You'll need to save the tags first either individually or use `PFObject.saveAll(inBackground: tags)` preferably before saving the post object. 

The same is true for quarrying. Doing `query.includeKey("tags")` includes PFObject pointers but  does not include PFRelations objects. Instead you can use `relation.query()` that returns a PFQuery for retrieving them. 

I will say I would like to see this changed as it really is a pain in comparison to saving new objects as points.

**Comment by jaysonng** [comment]
Thanks @noobs2ninjas . you got it right. PFRelation to a tags relation column. 

 I worked around this by using cloudcode  beforesave(Post) to save the tags before passing it on to the Post. But I do agree if it's possible to have this as standard behavior on relations as well.

Maybe we can tag this as a feature request ?

**Comment by noobs2ninjas** [comment]
Definitely. I've wanted this myself for quite a while. Let see if we can make it happen.
